tensornetwork/tensornetwork.py

```diff
@@ -308,19 +309,43 @@ def update_axis(self, old_axis: int, old_node: Node, new_axis: int,
     """
     if self.axis1 == old_axis and self.node1 is old_node:
       self.axis1 = new_axis
-      self.node1 = new_node
+      self.set_node1(new_node)
     elif self.axis2 == old_axis and self.node2 is old_node:
       self.axis2 = new_axis
-      self.node2 = new_node
+      self.set_node2(new_node)
     else:
       raise ValueError("Edge '{}' did not contain node '{}' on axis {}. "
                        "node1: '{}', axis1: {}, node2: '{}', axis2: {}".format(
                            self, old_node, old_axis, self.node1, self.axis1,
                            self.node2, self.axis2))
 
-  def is_dangling(self):
+  def set_node1(self, node: Node) -> None:
```

Comment:
You should be able to avoid breaking change to the API by using the `@*.setter` [decorator](https://docs.python.org/3/library/functions.html#property).

Reply:
Oh sweet. I didn't know that existed.